orderBook.js

```diff
@@ -0,0 +1,45 @@
+
+
+const isMatchingOrder = (existingBook, incomingOrder) => {
+  if (existingBook.type !== incomingOrder.type) {
+    if (incomingOrder.type === 'sell') {
+      if (existingBook.price >= incomingOrder.price) {
+        return true
+      }
+    } else if (existingBook.price <= incomingOrder.price) {
+      return true
+    }
+  }
+  return false
```

Comment:
This could be a lot cleaner. I would even argue it should be two functions since it is checking a couple of things.